packages/select/src/Select/components/Option/Option.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { HTMLAttributes } from "react"
+import { useFocusRing } from "@react-aria/focus"
+import { useOption } from "@react-aria/listbox"
+import { mergeProps } from "@react-aria/utils"
+import { Node } from "@react-types/shared"
+import classNames from "classnames"
+import { OverrideClassName } from "@kaizen/component-base"
+import { Icon } from "@kaizen/component-library"
+import check from "@kaizen/component-library/icons/check.icon.svg"
+import { useSelectionContext } from "../../provider"
+import { ItemType } from "../../types"
+import styles from "./Option.module.scss"
+
+export interface OptionProps
+  extends OverrideClassName<HTMLAttributes<HTMLDivElement>> {
+  item: Node<ItemType>
+}
+
+export const Option: React.VFC<OptionProps> = ({ item, classNameOverride }) => {
+  const { selectionState: state } = useSelectionContext()
+  // Get props for the option element
+  const ref = React.createRef<HTMLLIElement>()
```

Comment:
My understanding is that this would create a new ref on re-render, it might be more advisable to use useRef() instead

Reply:
I'm thinking that we can't have a null Ref passed to the component, and that's why we need to create one?

Reply:
https://beta.reactjs.org/apis/react/createRef#createref new doc is recommending against using createRef for function components, you can still create a ref with useRef()

`const ref = React.useRef< HTMLLIElement >(null);` should work here